numba/cuda/cudadrv/nvvm.py

```diff
@@ -423,25 +423,23 @@ def get(self):
     br label %lt_check
 
 done:
-    ; Return max
-    %ret = phi {T} [ %ptrval, %entry ], [ %dold, %lt_check ]
-    ret {T} %ret
+    ret {T} %ptrval
 }}
 """
 
 
 ir_numba_atomic_min = """
-define internal {T} @___numba_atomic_{T}_min({T}* %ptr, {T} %val) alwaysinline{{
+define internal {T} @___numba_atomic_{T}_{NAN}min({T}* %ptr, {T} %val) alwaysinline{{
 entry:
     %ptrval = load volatile {T}, {T}* %ptr
     ; Check if either value is a NaN and return *ptr early if so
-    %valnan = fcmp uno {T} %val, %ptrval
+    %valnan = fcmp uno {T} %val, %{PTR_OR_VAL}val
     br i1 %valnan, label %done, label %gt_check
 
 gt_check:
     %dold = phi {T} [ %ptrval, %entry ], [ %dcas, %attempt ]
     ; Continue attempts if dold > val
-    %lt = fcmp nnan ogt {T} %dold, %val
+    %lt = fcmp {OP} {T} %dold, %val
```

Comment:
```suggestion
    %gt = fcmp {OP} {T} %dold, %val
```
should probably be called `gt`? But moot/ call it `cmp`? if the templates are unified.